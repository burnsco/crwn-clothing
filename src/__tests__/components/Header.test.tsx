import "@testing-library/jest-dom"
import { render } from "@utils/testUtils"

const useRouter = jest.spyOn(require("next/router"), "useRouter")

describe("Header (navbar)", () => {
  useRouter.mockImplementation(() => ({
    route: "/shop/[category]",
    pathname: "/shop/[category]",
    query: { category: "sushi" },
    asPath: "/shop/sushi"
  }))
  it("Renders logo", async () => {
    const { getByTestId } = render(<h1>testing header</h1>)
    expect(getByTestId("nav-logo")).toBeInTheDocument()
  })

  it("Renders NavMenu", async () => {
    const { getByText } = render(<h1>testing header</h1>)
    expect(getByText(/shop/i)).toBeInTheDocument()
  })

  it("Renders Cart", async () => {
    const { getByText } = render(<h1>testing header</h1>)
    expect(getByText(/0/i)).toBeInTheDocument()
  })

  it("toggle dark mode", async () => {
    const { getByTestId } = render(<h1>testing header</h1>)
    expect(getByTestId("nav-logo")).toBeInTheDocument()
  })
})
