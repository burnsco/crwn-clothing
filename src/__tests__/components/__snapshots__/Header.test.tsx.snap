// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header (navbar) matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="css-wjqz34"
    id="navbar"
  >
    <nav
      aria-label="Primary Navigation"
      class="chakra-stack css-1kcbh8j"
    >
      <a
        aria-label="Home Page Link"
        class="chakra-link css-i3byiv"
        href="/"
      >
        <span
          class="css-1t6y3iw"
        >
          QUICK
        </span>
         EATS
      </a>
      <button
        aria-controls="menu-list-7"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-button chakra-menu__menu-button css-16n958m"
        id="menu-button-7"
        type="button"
      >
        <span
          class="chakra-button__icon css-ktvjz1"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="css-a9hmjk"
        >
          Sushi
        </span>
        <span
          class="chakra-button__icon css-1oeb2oe"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
            />
          </svg>
        </span>
      </button>
      <div
        class="css-r6z5ec"
        style="position: absolute; left: 0px; top: 0px; visibility: hidden;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-14dkbtu"
          id="menu-list-7"
          role="menu"
          style="transform-origin: top left; opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <div
            class="chakra-menu__group css-0"
            role="group"
          >
            <p
              class="chakra-menu__group__title css-1koowzy"
            >
              SHOP
            </p>
            <a
              class="chakra-menu__menuitem css-593qcw"
              href="/shop/burgers"
              id="menuitem-1"
              role="menuitem"
              tabindex="-1"
            >
              burgers
            </a>
            <a
              class="chakra-menu__menuitem css-593qcw"
              href="/shop/chicken"
              id="menuitem-2"
              role="menuitem"
              tabindex="-1"
            >
              chicken
            </a>
            <a
              class="chakra-menu__menuitem css-593qcw"
              href="/shop/pizza"
              id="menuitem-3"
              role="menuitem"
              tabindex="-1"
            >
              pizza
            </a>
            <a
              class="chakra-menu__menuitem css-593qcw"
              href="/shop/sushi"
              id="menuitem-4"
              role="menuitem"
              tabindex="-1"
            >
              sushi
            </a>
            <a
              class="chakra-menu__menuitem css-593qcw"
              href="/shop/sandwiches"
              id="menuitem-5"
              role="menuitem"
              tabindex="-1"
            >
              sandwiches
            </a>
          </div>
        </div>
      </div>
      <button
        class="chakra-button css-1qecutz"
        type="button"
      >
        0
        <span
          class="chakra-button__icon css-1oeb2oe"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M576 216v16c0 13.255-10.745 24-24 24h-8l-26.113 182.788C514.509 462.435 494.257 480 470.37 480H105.63c-23.887 0-44.139-17.565-47.518-41.212L32 256h-8c-13.255 0-24-10.745-24-24v-16c0-13.255 10.745-24 24-24h67.341l106.78-146.821c10.395-14.292 30.407-17.453 44.701-7.058 14.293 10.395 17.453 30.408 7.058 44.701L170.477 192h235.046L326.12 82.821c-10.395-14.292-7.234-34.306 7.059-44.701 14.291-10.395 34.306-7.235 44.701 7.058L484.659 192H552c13.255 0 24 10.745 24 24zM312 392V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24zm112 0V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24zm-224 0V280c0-13.255-10.745-24-24-24s-24 10.745-24 24v112c0 13.255 10.745 24 24 24s24-10.745 24-24z"
            />
          </svg>
        </span>
      </button>
      <button
        aria-label="Switch to dark mode"
        class="chakra-button css-ess5sq"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
          />
        </svg>
      </button>
    </nav>
  </header>
  <div
    class="chakra-container css-1au2bwr"
  >
    <div
      class="simple-grid-shop css-1vfkk4i"
    >
      <div
        class="css-bzb5yl"
      >
        <div
          class="css-c4fgdl"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="image-undefined"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fsushi%2F18&w=3840&q=75"
              srcset="/_next/image?url=%2Fsushi%2F18&w=640&q=75 640w, /_next/image?url=%2Fsushi%2F18&w=750&q=75 750w, /_next/image?url=%2Fsushi%2F18&w=828&q=75 828w, /_next/image?url=%2Fsushi%2F18&w=1080&q=75 1080w, /_next/image?url=%2Fsushi%2F18&w=1200&q=75 1200w, /_next/image?url=%2Fsushi%2F18&w=1920&q=75 1920w, /_next/image?url=%2Fsushi%2F18&w=2048&q=75 2048w, /_next/image?url=%2Fsushi%2F18&w=3840&q=75 3840w"
              style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </div>
        </div>
        <div
          class="css-q5fqw0"
        >
          <div
            class="chakra-stack css-149v5c4"
          >
            <span
              class="chakra-badge css-slmp4j"
            >
              $8
            </span>
            <h3
              class="css-1kygjsm"
            >
              Salmon Sashimi
            </h3>
          </div>
          <button
            class="chakra-button css-19nkbrb"
            spacing="6"
            type="button"
          >
            <span
              class="chakra-button__icon css-ktvjz1"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Add To Cart
          </button>
        </div>
      </div>
      <div
        class="css-bzb5yl"
      >
        <div
          class="css-c4fgdl"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="image-undefined"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fsushi%2F19&w=3840&q=75"
              srcset="/_next/image?url=%2Fsushi%2F19&w=640&q=75 640w, /_next/image?url=%2Fsushi%2F19&w=750&q=75 750w, /_next/image?url=%2Fsushi%2F19&w=828&q=75 828w, /_next/image?url=%2Fsushi%2F19&w=1080&q=75 1080w, /_next/image?url=%2Fsushi%2F19&w=1200&q=75 1200w, /_next/image?url=%2Fsushi%2F19&w=1920&q=75 1920w, /_next/image?url=%2Fsushi%2F19&w=2048&q=75 2048w, /_next/image?url=%2Fsushi%2F19&w=3840&q=75 3840w"
              style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </div>
        </div>
        <div
          class="css-q5fqw0"
        >
          <div
            class="chakra-stack css-149v5c4"
          >
            <span
              class="chakra-badge css-slmp4j"
            >
              $5
            </span>
            <h3
              class="css-1kygjsm"
            >
              California Roll
            </h3>
          </div>
          <button
            class="chakra-button css-19nkbrb"
            spacing="6"
            type="button"
          >
            <span
              class="chakra-button__icon css-ktvjz1"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Add To Cart
          </button>
        </div>
      </div>
      <div
        class="css-bzb5yl"
      >
        <div
          class="css-c4fgdl"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="image-undefined"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fsushi%2F20&w=3840&q=75"
              srcset="/_next/image?url=%2Fsushi%2F20&w=640&q=75 640w, /_next/image?url=%2Fsushi%2F20&w=750&q=75 750w, /_next/image?url=%2Fsushi%2F20&w=828&q=75 828w, /_next/image?url=%2Fsushi%2F20&w=1080&q=75 1080w, /_next/image?url=%2Fsushi%2F20&w=1200&q=75 1200w, /_next/image?url=%2Fsushi%2F20&w=1920&q=75 1920w, /_next/image?url=%2Fsushi%2F20&w=2048&q=75 2048w, /_next/image?url=%2Fsushi%2F20&w=3840&q=75 3840w"
              style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </div>
        </div>
        <div
          class="css-q5fqw0"
        >
          <div
            class="chakra-stack css-149v5c4"
          >
            <span
              class="chakra-badge css-slmp4j"
            >
              $7
            </span>
            <h3
              class="css-1kygjsm"
            >
              Cooked Ramen
            </h3>
          </div>
          <button
            class="chakra-button css-19nkbrb"
            spacing="6"
            type="button"
          >
            <span
              class="chakra-button__icon css-ktvjz1"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Add To Cart
          </button>
        </div>
      </div>
      <div
        class="css-bzb5yl"
      >
        <div
          class="css-c4fgdl"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="image-undefined"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fsushi%2F21&w=3840&q=75"
              srcset="/_next/image?url=%2Fsushi%2F21&w=640&q=75 640w, /_next/image?url=%2Fsushi%2F21&w=750&q=75 750w, /_next/image?url=%2Fsushi%2F21&w=828&q=75 828w, /_next/image?url=%2Fsushi%2F21&w=1080&q=75 1080w, /_next/image?url=%2Fsushi%2F21&w=1200&q=75 1200w, /_next/image?url=%2Fsushi%2F21&w=1920&q=75 1920w, /_next/image?url=%2Fsushi%2F21&w=2048&q=75 2048w, /_next/image?url=%2Fsushi%2F21&w=3840&q=75 3840w"
              style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </div>
        </div>
        <div
          class="css-q5fqw0"
        >
          <div
            class="chakra-stack css-149v5c4"
          >
            <span
              class="chakra-badge css-slmp4j"
            >
              $7
            </span>
            <h3
              class="css-1kygjsm"
            >
              Avocado Roll
            </h3>
          </div>
          <button
            class="chakra-button css-19nkbrb"
            spacing="6"
            type="button"
          >
            <span
              class="chakra-button__icon css-ktvjz1"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Add To Cart
          </button>
        </div>
      </div>
      <div
        class="css-bzb5yl"
      >
        <div
          class="css-c4fgdl"
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="image-undefined"
              decoding="async"
              sizes="100vw"
              src="/_next/image?url=%2Fsushi%2F22&w=3840&q=75"
              srcset="/_next/image?url=%2Fsushi%2F22&w=640&q=75 640w, /_next/image?url=%2Fsushi%2F22&w=750&q=75 750w, /_next/image?url=%2Fsushi%2F22&w=828&q=75 828w, /_next/image?url=%2Fsushi%2F22&w=1080&q=75 1080w, /_next/image?url=%2Fsushi%2F22&w=1200&q=75 1200w, /_next/image?url=%2Fsushi%2F22&w=1920&q=75 1920w, /_next/image?url=%2Fsushi%2F22&w=2048&q=75 2048w, /_next/image?url=%2Fsushi%2F22&w=3840&q=75 3840w"
              style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
          </div>
        </div>
        <div
          class="css-q5fqw0"
        >
          <div
            class="chakra-stack css-149v5c4"
          >
            <span
              class="chakra-badge css-slmp4j"
            >
              $18
            </span>
            <h3
              class="css-1kygjsm"
            >
              Red Dragon Roll
            </h3>
          </div>
          <button
            class="chakra-button css-19nkbrb"
            spacing="6"
            type="button"
          >
            <span
              class="chakra-button__icon css-ktvjz1"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Add To Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
